xarray/tests/test_backends.py

```diff
@@ -2189,15 +2224,44 @@ def test_platecarree(self):
                                          })
             with xr.open_rasterio(tmp_file) as rioda:
                 assert_allclose(rioda, expected)
-                assert 'crs' in rioda.attrs
                 assert isinstance(rioda.attrs['crs'], basestring)
-                assert 'res' in rioda.attrs
                 assert isinstance(rioda.attrs['res'], tuple)
-                assert 'is_tiled' in rioda.attrs
                 assert isinstance(rioda.attrs['is_tiled'], np.uint8)
-                assert 'transform' in rioda.attrs
                 assert isinstance(rioda.attrs['transform'], tuple)
 
+    def test_notransform(self):
+        # regression test for https://github.com/pydata/xarray/issues/1686
+        import rasterio
+        import warnings
+
+        # Create a geotiff file
+        with warnings.catch_warnings():
+            # This warning is expected
+            warnings.filterwarnings('ignore', category=UserWarning)
```

Comment:
which specific warning is expected here?

`pytest.assert_warns()` might be more appropriate, if we want to test the warning as part of the API

Reply:
It's rasterio which sends a warning here. I explain it in more detail know, and catch only this specific warning.